lib/geoservices/queryable.rb

```diff
@@ -14,7 +14,16 @@ def query(layer_idx=0, options={})
     # returns nil if the layer does not have the Capability: Query
     def count(layer_idx=0, options={})
       return nil unless self.metadata["capabilities"] =~ /Query/
-      query(layer_idx, options.merge(:returnCountOnly => true))
+      # Current Version is not returned at lower than 10.1
+      unless self.metadata['currentVersion'].nil?
```

Comment:
to reiterate, returnCountOnly and currentVersion on layers were added at the same time.